python_modules/libraries/dagster-azure/dagster_azure/adls2/io_manager.py

```diff
@@ -25,6 +25,8 @@
 
 
 class PickledObjectADLS2IOManager(UPathIOManager):
+    extension = ".pkl"
```

Comment:
I think this needs to be an opt-in bit of config for the IO Manager - otherwise we will fail to do reads for past writes that used the uri without the extension. For op/job users its unlikely to be reading old URIs but for assets it can be very common.

Reply:
Thanks @alangenfeld for the feedback! I've made the use of the extension optional via the io_manager config. Curious what you think